explorer/testdata/pointer/class_lifetime.carbon

```diff
@@ -0,0 +1,33 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{explorer-run}
+// RUN: %{explorer-run-trace}
+// CHECK:STDOUT: Before New A
+// CHECK:STDOUT: CONSTRUCTOR A
+// CHECK:STDOUT: DESTRUCTOR A
```

Comment:
This test was meant to document the lifecycle of an object with `New`/`Delete` mainly, will drop before we merge.